### PR TITLE
chore(azure): allow passing custom names and tags to all resources

### PR DESCRIPTION
**Pull request by swiknaba**
#### Summary

Allow passing custom names & tags to all Azure modules.

Tags & names always have a default name following the convention of this repo (`${project}-${environment}`).

### CODE REVIEW

File: azure/blob-storage/main.tf [hunk 1]
```diff
@@ -55,7 +55,7 @@ resource "azurerm_storage_account" "main" {
     }
   }
 
-  tags = local.default_tags
+  tags = coalesce(var.tags, local.default_tags)
```
Comment (samkahchiin) [inline azure/blob-storage/main.tf:58]: Just an idea, WDYT if we use `merge` to keep the default_tags if it is not override?

Comment (swiknaba) [reply azure/blob-storage/main.tf:58]: I'm conflicted. What if users want to use **exactly** their custom tags, and won't expect any other tags to be added?

Comment (samkahchiin) [reply azure/blob-storage/main.tf:58]: Good point.

File: azure/container-registry/variables.tf [hunk 2]
```diff
@@ -66,8 +62,20 @@ variable "sku" {
...
+
+variable "name" {
+  type        = string
+  default     = null
+  description = "Defaults to 'project-environment'."
+}
+
 locals {
-  name = "${var.project}-${var.environment}"
+  default_name = "${var.project}-${var.environment}"
```
Comment (samkahchiin) [inline azure/container-registry/variables.tf:77]: Container registry name cannot contain special cahracter

**Review by samkahchiin — CHANGES_REQUESTED** [review]

File: azure/container-registry/main.tf [hunk 3]
```diff
@@ -1,5 +1,5 @@
 resource "azurerm_container_registry" "main" {
-  name                          = var.name == null ? var.project : var.name
+  name                          = coalesce(var.name, local.default_name)
```
Comment (samkahchiin) [inline azure/container-registry/main.tf:2]: where do we get the default_name local here?

**Review by samkahchiin — CHANGES_REQUESTED** [review]

**Review by samkahchiin — APPROVED** [review]

**Pull request merged**